src/ansys/mapdl/core/xpl.py

```diff
@@ -36,7 +35,9 @@
 
 def id_generator(size=6, chars=string.ascii_uppercase):
     """Generate a random string using only uppercase letters."""
-    return "".join(random.choice(chars) for _ in range(size))
+    import secrets
+
+    return "".join(secrets.choice(chars) for _ in range(size))
```

Comment:
``secrets`` is intended for cryptographical applications where it is important to have totally random numbers (not predictable). Here it is not that important. What could be the problem if an attacker guess where the next directory is going to be? 

It could be interesting if we were supporting any sort of authentication with the MAPDL server... but anyone can get in our connection, so... it is not that important at the moment. 

Keep the change, but I dont think it was really a vulnerability.

Reply:
I agree with you